ada_feeding_perception/ada_feeding_perception/food_on_fork.py

```diff
@@ -0,0 +1,363 @@
+                    ParameterDescriptor(
+                        name="test",
+                        type=ParameterType.PARAMETER_BOOL,
+                        description="testing",
+                        read_only=True,
+                    ),
+                ),
+            ],
+        )
+
+    def normalize_to_uint8(self, img):
+        """
+        Normalize the image to 0-255
+
+        Parameters:
+        ----------
+        img: image to normalize
+        """
+        img_normalized = ((img - img.min()) / (img.max() - img.min()) * 255).astype(
+            "uint8"
+        )
+        return img_normalized
+
+    def listener_callback_depth(self, depth_img_msg: Image) -> None:
+        """
+        Calculates and publishes probability indicating whether the provided depth image has food
+        on fork
+
+        Parameters:
+        ----------
+        depth_img_msg: depth image message from the subscriber
+        """
+        depth_img = None
+        try:
+            depth_img = self.bridge.imgmsg_to_cv2(depth_img_msg, "passthrough")
+        except CvBridgeError as e:
+            print(e)
+
+        if depth_img is not None:
+            # VVVV---BELOW is the code when using single feature of only the NUMBER OF PIXELS---VVVV
+            # num_pixels = self.food_on_fork_num_pixels(depth_img)
+            # if self.test:
+            #     self.get_logger().info("num pixels: " + str(num_pixels))
+            # prediction = self.predict_food_on_fork(num_pixels)
+            # if self.test:
+            #     self.get_logger().info("Prediction Probability: " + str(prediction))
+            # ^^^^---ABOVE is the code when using single feature of only the NUMBER OF PIXELS---^^^^
+
+            # VVVV---BELOW is the code when using EACH PIXEL as a feature---VVVV
+            depth_img_copy = np.copy(depth_img)
+
+            # Crop the depth image to the specified dimensions
+            left_top_x, left_top_y = self.left_top_corner
+            right_bottom_x, right_bottom_y = self.right_bottom_corner
+            cropped_img = depth_img_copy[
+                left_top_y:right_bottom_y, left_top_x:right_bottom_x
+            ]
+            # cv.imshow("cropped_img", self.normalize_to_uint8(cropped_img))
+
+            # Pre-process the image such that if the depth values are within the specified
+            # frustum, then those pixels are converted to be 1 and if they are not within the
+            # frustum, then those pixels are converted to be 0
+            img_converted = np.where(
+                np.logical_or(cropped_img < self.min_dist, cropped_img > self.max_dist),
+                0,
+                1,
+            ).astype("uint8")
+            cropped_img_np = np.array(img_converted)
+
+            # self.get_logger().info(str(cropped_img_np))
+            # self.get_logger().info(str(np.count_nonzero(cropped_img_np == 1)))
+            # cv.imshow("cropped", cropped_img_np * 255)
+            # cv.waitKey(40)
```

Comment:
It is fine to keep commented out debug statements, but only keep the most important that you actually think a future programmer may need to comment out. Of these 4 lines, decide which is the most important and just keep that. (Or delete them all)